.github/workflows/test-integration-17.yml

```diff
@@ -0,0 +1,37 @@
+name: Integration@node 17
+on:
+  pull_request:
+  push:
+    branches-ignore:
+      - 'gh-pages'
+
+jobs:
+  test-integration-17:
+    runs-on: ubuntu-latest
+
+    services:
+      redis:
+        image: redis
+        options: >-
+          --health-cmd "redis-cli ping"
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+        ports:
+          - 6379:6379
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Setup
+        uses: ./.github/actions/setup
+        with:
+          node-version: 17
+        env:
+          NPM_CONFIG_ENGINE_STRICT: 'false'
+
+      - name: Integration Tests
+        uses: ./.github/actions/integration-tests
+        with:
+          github-token: '${{ secrets.GITHUB_TOKEN }}'
```

Comment:
Even when the PR is submitted from a fork, the build is assigned a short-lived read-only token which is enough for running the tests.